src/client.ts

```diff
@@ -27,80 +28,244 @@ import type { Browsers } from "./types.ts";
    */
-  public async done() {
-    await this.#protocol.done();
+  public _popPage(pageTargetId: string) {
+    this.#pages = this.#pages.filter((page) => page.target_id !== pageTargetId);
   }
 
   /**
-   * Go to a specific page
+   * For internal use.
    *
-   * @param urlToVisit - The page to go to
+   * Pushed a new item to the pages array
    *
-   * @returns A page instance, with methods to help you directly interact with the page
+   * @param page - Page to push
    */
-  public async goTo(urlToVisit: string): Promise<Page> {
-    const method = "Page.loadEventFired";
-    this.#protocol.notification_resolvables.set(method, deferred());
-    const notificationPromise = this.#protocol.notification_resolvables.get(
-      method,
-    );
-    const res = await this.#protocol.sendWebSocketMessage<
-      ProtocolTypes.Page.NavigateRequest,
-      ProtocolTypes.Page.NavigateResponse
-    >(
-      "Page.navigate",
-      {
-        url: urlToVisit,
-      },
-    );
-    await notificationPromise;
-    if (res.errorText) {
-      await this.#protocol.done(
-        `${res.errorText}: Error for navigating to page "${urlToVisit}"`,
+  public _pushPage(
+    page: Page,
+  ): void {
+    this.#pages.push(page);
+  }
+
+  /**
+   * A way to get a page. Useful if a new tab/page has opened
+   *
+   * @example
+   * ```js
+   * const { browser, page } = await buildFor("chrome");
+   * console.log(await browser.page(1)); // Your initial page, exactly what `page` above is
+   * // You middle click an element
+   * console.log(await browser.page(2)); // will return a Page representation of the newly opened page
+   * ```
+   *
+   * @param pageNumber - Which page to get, the first/initial page? 1. The second you just opened via a click? 2.
+   * @returns The page
+   */
+  public async page(pageNumber: number): Promise<Page> {
+    // `i` is given to us in a way that makes the user understand exactly what page they want.
+    // If 1, they want the first page, so we will get the 0th index
+    const index = pageNumber - 1;
+
+    if (!this.#pages[index]) {
+      await this.close(
+        "You have request to get page number " + pageNumber + ", but only " +
+          this.#pages.length +
+          " pages are opened. If the issue persists, please submit an issue.",
+        RangeError,
       );
     }
-    return new Page(this.#protocol);
+
+    return this.#pages[index];
+  }
+
+  /**
+   * Close/stop the sub process, and close the ws connection. Must be called when finished with all your testing
+   *
+   * @param errMsg - If provided, after closing, will throw an error with the message. Useful for throwing errors but making sure all resources are closed beforehand
+   * @param errClass - The class name to throw with the error message, eg `SyntaxError`. Defaults to `Error`
+   */
+  public async close(
+    errMsg?: string,
+    // deno-lint-ignore no-explicit-any
+    errClass: { new (message: string): any } = Error,
```

Comment:
could `any` be `Error` since `new (message :string): any` would essentially be an extension an `Error` ? or is there a case where it would not be an `Error` ?

Reply:
`SyntaxError`, `INvalidArgumentError`, `OutOfBoundsError` etc, its really just any of the error classes so you can do

```
if (options.start > 100) {
  // browser will break in this case
  await this.close("Not allowed to be greater than 100!", InvalidLengthError)
}
```

Reply:
wondering if we can ditch the `any` and just use `Error` then? although that would require users to extend `Error` whereas Drash has `IErrorHandler` logic that allows users to pass in a random object that just has to match the interface and not necessarily extend `ErrorHandler`

Reply:
actually you're onto something, i can just do `typeof Error`, because all error objects extend `Error` class